libsoftwarecontainer/src/gateway/gateway.cpp

```diff
@@ -100,7 +101,12 @@ bool Gateway::activate() {
 }
 
 bool Gateway::teardown() {
-    if (GatewayState::ACTIVATED != m_state) {
+    /* If a gatweway is dynamic it's possible to do teardown if
+       it has been activated at least once before.
+
+       If a gateway is not dynamic it's possible to do teardown if
+       it is in state ACTIVATED */
+    if ((m_isDynamic && !m_activatedOnce) || (!m_isDynamic && GatewayState::ACTIVATED != m_state)) {
```

Comment:
can we rephare this like 
`if (GatewayState::ACTIVATED != m_state && !m_activatedOnce)`